I am super confused as to why this e2e test is failing with 

```
  1 failing

  1) Loki query builder
       should be able to use all modes:
     AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-testid="data-testid Select label"]`, but never found it.
      at Context.eval (webpack://grafana/./e2e/various-suite/loki-query-builder.spec.ts:69:46)
```

The selector is there, the aria-label has been switched to data-testid, the selector in the component is written like the others 🤷🏻. @grafana/observability-logs do you have any idea why this might be? I must have missed something somewhere. If you have some extra eyes to help with this that would be appreciated 🙏🏻 

https://github.com/grafana/grafana/blob/6f3febdb6077ffb4f8194aff849cf5d22c7d34c0/public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx#L79

<img width="1094" alt="Screenshot 2023-11-22 at 5 57 14 PM" src="https://github.com/grafana/grafana/assets/25674746/ba3b3e4e-dd7d-490b-ac03-e34d7c087efa">

--- a/e2e/various-suite/loki-query-builder.spec.ts
+++ b/e2e/various-suite/loki-query-builder.spec.ts
@@ -69,11 +69,20 @@ describe('Loki query builder', () => {
     e2e.components.QueryBuilder.labelSelect().should('be.visible').click();
     // wait until labels are loaded and set on the component before starting to type
     cy.wait('@labelsRequest');
-    e2e.components.QueryBuilder.labelSelect().type('instance{enter}');
-    e2e.components.QueryBuilder.matchOperatorSelect().should('be.visible').click().type('=~{enter}');
+    e2e.components.QueryBuilder.labelSelect().children('div').children('input').type('instance{enter}');
+    e2e.components.QueryBuilder.matchOperatorSelect()
+      .should('be.visible')
+      .click()
+      .children('div')
+      .children('input')
+      .type('=~{enter}', { force: true });
     e2e.components.QueryBuilder.valueSelect().should('be.visible').click();
     cy.wait('@valuesRequest');
-    e2e.components.QueryBuilder.valueSelect().type('instance1{enter}').type('instance2{enter}');
+    e2e.components.QueryBuilder.valueSelect()
+      .children('div')
+      .children('input')
+      .type('instance1{enter}')
+      .type('instance2{enter}');
     cy.contains(MISSING_LABEL_FILTER_ERROR_MESSAGE).should('not.exist');
     cy.contains(finalQuery).should('be.visible');
 

--- a/packages/grafana-ui/src/components/Select/SelectBase.tsx
+++ b/packages/grafana-ui/src/components/Select/SelectBase.tsx
@@ -73,6 +73,7 @@ export function SelectBase<T, Rest = {}>({
   allowCustomValue = false,
   allowCreateWhileLoading = false,
   'aria-label': ariaLabel,
+  'data-testid': dataTestid,
   autoFocus = false,
   backspaceRemovesValue = true,
   blurInputOnSelect,
@@ -199,6 +200,7 @@ export function SelectBase<T, Rest = {}>({
 
   const commonSelectProps = {
     'aria-label': ariaLabel,
+    'data-testid': dataTestid,
     autoFocus,
     backspaceRemovesValue,
     blurInputOnSelect,

--- a/packages/grafana-ui/src/components/Select/ValueContainer.tsx
+++ b/packages/grafana-ui/src/components/Select/ValueContainer.tsx
@@ -29,17 +29,21 @@ class UnthemedValueContainer extends Component<any & { theme: GrafanaTheme2 }> {
   }
 
   renderContainer(children?: ReactNode) {
-    const { isMulti, theme } = this.props;
+    const { isMulti, theme, selectProps } = this.props;
     const noWrap = this.props.selectProps?.noMultiValueWrap && !this.props.selectProps?.menuIsOpen;
     const styles = getSelectStyles(theme);
-
+    const dataTestid = selectProps['data-testid'];
     const className = cx(
       styles.valueContainer,
       isMulti && styles.valueContainerMulti,
       noWrap && styles.valueContainerMultiNoWrap
     );
 
-    return <div className={className}>{children}</div>;
+    return (
+      <div data-testid={dataTestid} className={className}>
+        {children}
+      </div>
+    );
   }
 }
 

--- a/packages/grafana-ui/src/components/Select/types.ts
+++ b/packages/grafana-ui/src/components/Select/types.ts
@@ -18,6 +18,7 @@ export type LoadOptionsCallback<T> = (options: Array<SelectableValue<T>>) => voi
 export interface SelectCommonProps<T> {
   /** Aria label applied to the input field */
   ['aria-label']?: string;
+  ['data-testid']?: string;
   allowCreateWhileLoading?: boolean;
   allowCustomValue?: boolean;
   /** Focus is set to the Select when rendered*/

